source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml

```diff
@@ -266,8 +266,25 @@
         >
         Firebase.Firestore.Model.Value.FieldValue
     </attr>
-        
-        
+    <attr
+        path="/api/package[@name='com.google.firestore.v1']/class[@name='FirestoreGrpc.FirestoreStub']/method[@name='build' and count(parameter)=2 and parameter[1][@type='io.grpc.Channel'] and parameter[2][@type='io.grpc.CallOptions']]"

```

Comment:
What error does the new `metadata` changes fix?  It looks like they worked previously without these changes, and changing them will cause API breakages:

(same for inappmessaging)

#### Type Changed: Google.Firestore.V1.FirestoreGrpc.FirestoreBlockingStub

Removed method:

```csharp
protected FirestoreGrpc.FirestoreBlockingStub Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

Added method:

```csharp
protected override Java.Lang.Object Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```


#### Type Changed: Google.Firestore.V1.FirestoreGrpc.FirestoreFutureStub

Removed method:

```csharp
protected FirestoreGrpc.FirestoreFutureStub Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

Added method:

```csharp
protected override Java.Lang.Object Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```


#### Type Changed: Google.Firestore.V1.FirestoreGrpc.FirestoreStub

Removed method:

```csharp
protected FirestoreGrpc.FirestoreStub Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

Added method:

```csharp
protected override Java.Lang.Object Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

Reply:
without this:

```
    <attr
        path="/api/package[@name='com.google.firestore.v1']/class[@name='FirestoreGrpc.FirestoreStub']/method[@name='build' and count(parameter)=2 and parameter[1][@type='io.grpc.Channel'] and parameter[2][@type='io.grpc.CallOptions']]"
        name="managedReturn"
        >
        Java.Lang.Object
    </attr>
```

the error is:

```
./generated/com.google.firebase.firebase-firestore/obj/Release/monoandroid90/generated/src/Google.Firestore.V1.FirestoreGrpc.cs(941,31): error CS0534: 'FirestoreGrpc.FirestoreStub' does not implement inherited abstract member 'AbstractStub.Build(Channel, CallOptions)' [./generated/com.google.firebase.firebase-firestore/com.google.firebase.firebase-firestore.csproj]
```

Reply:
besides those methods are `protected`, so API change is not that critical.

Reply:
The Java is:
```java
public abstract class AbstractStub<S extends AbstractStub<S>> {
  protected abstract S build(Channel paramChannel, CallOptions paramCallOptions);
}
```

Looks like previously this method was not being bound, and now it is, likely due to a `generator` fix.

```csharp
[Register("build", "(Lio/grpc/Channel;Lio/grpc/CallOptions;)Lio/grpc/stub/AbstractStub;", "GetBuild_Lio_grpc_Channel_Lio_grpc_CallOptions_Handler")]
protected abstract Object Build(Channel p0, CallOptions p1);
```

I guess the breakage is ok, like you said it's `protected` API.

Reply:
> Looks like previously this method was not being bound, and now it is, likely due to a `generator` fix.

Yeah. I suspected that, but wasn't sure

Reply:
... and yes nobody touched these bindings for quite some time (and not gone deeper into analysis)